### PR TITLE
nsqadmin: improve docs for exposed metrics

### PR DESCRIPTION
**Pull request by jehiah**
fixes bitly/nsq#629

RFR @mreiferson

### CODE REVIEW

File: _posts/2012-10-01-nsqadmin.md [hunk 1]
```diff
@@ -56,3 +58,35 @@ by setting `--notification-http-endpoint` to `http://addr.of.nsqd/put?topic=admi
...
 [gaproxy]: https://github.com/bitly/google_auth_proxy
 [statsdaemon]: https://github.com/bitly/statsdaemon
+
+### Metrics
+
+The following metrics are exposed through `nsqadmin` on Topic, Channel and Client Connections.
+
+**Message Queues:**
+
+* **Depth:** Sum of message in memory + messages on disk. This is the count of messages a client can consume immediately.
```
Comment (mreiferson) [inline _posts/2012-10-01-nsqadmin.md]: > Sum of messages in memory + on disk (i.e. the "backlog" of messages pending delivery)

File: _posts/2012-10-01-nsqadmin.md [hunk 2]
```diff
@@ -56,3 +58,35 @@ by setting `--notification-http-endpoint` to `http://addr.of.nsqd/put?topic=admi
...
+* **In-Flight:** Count of messages sent to a client but not yet Finished (`FIN`), Requeued (`REQ`) or Timed Out.
+* **Deferred:** Count of messages that were Requeued or Timed Out and are not yet available for delivery to a client.
```
Comment (mreiferson) [inline _posts/2012-10-01-nsqadmin.md]: > Count of messages that were requeued or explicitly deferred and are not yet available for delivery

(timed out are immediately requeued)

File: _posts/2012-10-01-nsqadmin.md [hunk 3]
```diff
@@ -56,3 +58,35 @@ by setting `--notification-http-endpoint` to `http://addr.of.nsqd/put?topic=admi
...
 [statsdaemon]: https://github.com/bitly/statsdaemon
+
+### Metrics
+
+The following metrics are exposed through `nsqadmin` on Topic, Channel and Client Connections.
+
+**Message Queues:**
+
+* **Depth:** Sum of message in memory + messages on disk. This is the count of messages a client can consume immediately.
+* **In-Flight:** Count of messages sent to a client but not yet Finished (`FIN`), Requeued (`REQ`) or Timed Out.
```
Comment (mreiferson) [inline _posts/2012-10-01-nsqadmin.md]: I don't see any reason to capitalize Finished, Requeued, and Timed Out?

File: _posts/2012-10-01-nsqadmin.md [hunk 4]
```diff
@@ -56,3 +58,35 @@ by setting `--notification-http-endpoint` to `http://addr.of.nsqd/put?topic=admi
...
+* **Deferred:** Count of messages that were Requeued or Timed Out and are not yet available for delivery to a client.
+
+**Statistics:**
+
+* **Requeued:** Messages delivered to a client and subsequently moved to the deferred queue for later retry.
```
Comment (mreiferson) [inline _posts/2012-10-01-nsqadmin.md]: > Total count of messages that have been added back to the queue due to time outs or explicit requeues

File: _posts/2012-10-01-nsqadmin.md [hunk 5]
```diff
@@ -56,3 +58,35 @@ by setting `--notification-http-endpoint` to `http://addr.of.nsqd/put?topic=admi
...
+* **Timed Out:** Messages delivered to a client that didn't receive a response before configured timeout that were server-side requeued.
```
Comment (mreiferson) [inline _posts/2012-10-01-nsqadmin.md]: > Total count of messages that were requeued after not receiving a response from the client before the configured timeout

File: _posts/2012-10-01-nsqadmin.md [hunk 6]
```diff
@@ -56,3 +58,35 @@ by setting `--notification-http-endpoint` to `http://addr.of.nsqd/put?topic=admi
...
+
+### Metrics
+
+The following metrics are exposed through `nsqadmin` on Topic, Channel and Client Connections.
+
+**Message Queues:**
+
+* **Depth:** Sum of messages in memory + on disk (i.e. the "backlog" of messages pending delivery).
+* **In-Flight:** Current Count of messages sent to a client but not yet finished (`FIN`), requeued (`REQ`) or timed out.
+* **Deferred:** Count of messages that were requeued and explicitly deferred which are not yet available for delivery.
```
Comment (mreiferson) [inline _posts/2012-10-01-nsqadmin.md]: prefix with "Current"

File: _posts/2012-10-01-nsqadmin.md [hunk 7]
```diff
@@ -56,3 +58,35 @@ by setting `--notification-http-endpoint` to `http://addr.of.nsqd/put?topic=admi
...
 [gaproxy]: https://github.com/bitly/google_auth_proxy
 [statsdaemon]: https://github.com/bitly/statsdaemon
+
+### Metrics
+
+The following metrics are exposed through `nsqadmin` on Topic, Channel and Client Connections.
+
+**Message Queues:**
+
+* **Depth:** Sum of messages in memory + on disk (i.e. the "backlog" of messages pending delivery).
```
Comment (mreiferson) [inline _posts/2012-10-01-nsqadmin.md]: also prefix with "Current"

File: _posts/2012-10-01-nsqadmin.md [hunk 8]
```diff
@@ -56,3 +58,35 @@ by setting `--notification-http-endpoint` to `http://addr.of.nsqd/put?topic=admi
...
+* **In-Flight:** Current Count of messages sent to a client but not yet finished (`FIN`), requeued (`REQ`) or timed out.
```
Comment (mreiferson) [inline _posts/2012-10-01-nsqadmin.md]: `s/sent to a client/delivered`

File: _posts/2012-10-01-nsqadmin.md [hunk 9]
```diff
@@ -56,3 +58,35 @@ by setting `--notification-http-endpoint` to `http://addr.of.nsqd/put?topic=admi
...
+* **Connections:** Current number of connected clients.
+
+**Client Connections:**
+
+* **Client Host:** Client ID (hostname) and on-hover the connection remote-address.
+* **Protocol:** NSQ protocol version and client user-agent.
+* **Attributes:** TLS and AUTH connection state.
+* **NSQd Host:** Address of the nsqd node connected to.
+* **In-flight:** Number of messages sent on a connection which are not yet Finished or Requeued.
+* **Ready Count:** Max number of messages that can be in-flight on this connection. This is controlled by a clients max-in-flight setting.
```
Comment (mreiferson) [inline _posts/2012-10-01-nsqadmin.md]: `s/clients/client's`

Comment (mreiferson) [inline _posts/2012-10-01-nsqadmin.md]: add ticks around `max_in_flight`?

File: _posts/2012-10-01-nsqadmin.md [hunk 10]
```diff
@@ -56,3 +58,35 @@ by setting `--notification-http-endpoint` to `http://addr.of.nsqd/put?topic=admi
...
+* **Finished:** Sum of Finish (`FIN`) responses.
+* **Requeued:** Sum of Requeue (`REQ`) responses.
+* **Messages:** Count of messages sent by nsqd on a connection.
```
Comment (mreiferson) [inline _posts/2012-10-01-nsqadmin.md]: `s/by nsqd on a/to this`

**Comment by mreiferson** [comment]
:hammer:

**Comment by Dieterbe** [comment]
thanks a lot guys.
a few comments though:

1)
>  +* **Rate:** The per-second rate of new messages (available when graphite integration is enabled). 

is this per-second rate computed by looking at just the last second? it could also be some longer interval (and still adjusted to be per-second) and this is not really clear. or is it like a moving average?

2) For Timed Out and Requeued "Total count of messages..." should probably be "Total times a message..." because if the same message times out twice or gets requeued twice, it'll count for two right, even though it's the same message?

3) Under "client connections"

> **In-flight:** Number of messages sent on a connection which are not yet Finished or Requeued.

is this a total or current number? should probably be made explicit.

4) also under "client connections: finished, requeued and messages should probably also state "total count of", because presumably it's not a current count.

**Comment by mreiferson** [comment]
thanks @Dieterbe, updated in cb264a8

**Comment by Dieterbe** [comment]
great, thank you.

**Pull request merged**